rbc/omnisci_backend/omnisci_pipeline.py

```diff
@@ -54,6 +54,41 @@ def run_pass(self, state):
         return True  # we changed the IR
 
 
+@register_pass(mutates_CFG=False, analysis_only=False)
+class RewriteRaises(FunctionPass):
+    _name = "rewrite_raises"
+
+    def __init__(self):
+        FunctionPass.__init__(self)
+
+    def run_pass(self, state):
+        mutated = False
+        func_ir = state.func_ir
+        for blk in func_ir.blocks.values():
+            for _raise in blk.find_insts(ir.Raise):
+                name = 'table_function_error'
+                loc = _raise.loc
+                fn = ir.Assign(value=ir.Global(name, table_function_error, loc=loc),
+                               target=blk.scope.make_temp(loc=loc), loc=loc)
+
+                # replace called function by 'table_function_error'
+                call = blk.find_variable_assignment(_raise.exception.name)
+                glob = blk.find_variable_assignment(call.value.func.name)
+                blk.insert_after(fn, glob)
+                call.value.func = fn.target
+
+                # create a return inst
+                ret = ir.Return(value=call.target, loc=loc)
+                blk.append(ret)
+
+                # remove raise instruction
+                blk.remove(_raise)
+
```

Comment:
At first I was confused by this, because we remove the `_raise` in the middle of the block and we append the `return` at the end.
Then I realized that this works only when the `raise` is *always* the last instruction of the block. I think this is a property that we can assume as guaranteed, but it's better to add an `assert` just in case, before doing the `append/remove` dance:
```python
assert blk.body[-1] is _raise
```